packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -288,6 +301,82 @@ export interface AutoScalingGroupProps extends CommonAutoScalingGroupProps {
   readonly role?: iam.IRole;
 }
 
+/**
+ * Group metrics that an Auto Scaling group sends to Amazon CloudWatch.
+ */
+export class GroupMetric {
+
+  /**
+   * All available GroupMetrics.
+   */
+  public static readonly ALL = new Array<GroupMetric>();
```

Comment:
This can cause some weird/unexpected behaviour.

For instance, a user can 

```ts
const group = new autoscaling.AutoScalingGroup(stack, 'ASG', {
  groupMetricsCollections: [
    { metrics: GroupMetric.ALL, }
  ]
});

group.emitAllGroupMetrics();
```

or the other way around.

Do we handle this? I would expect that all metrics will be emitted here.

On the other hand if you turned the `ALL` into a special string, say `'[ALL]'`, and smartly handled this at the time of resolution, this might be a better experience.

Reply:
Yes this is handled, since `ALL` is an empty array this result it `metrics` being `undefined` so this two calls generate legal CF and ASG configuration:

```
     "MetricsCollection": [
          {
            "Granularity": "1Minute"
          },
          {
            "Granularity": "1Minute"
          }
        ],
```